yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,243 @@
+            'upload_date': '20211102'
+        }
+    }]
+
+    def _real_extract(self, original_url):
+        import datetime
+
+        video_id = self._match_id(original_url)
+        downloaded_json = self._download_json(url_or_request=self._JSON_BASE_URL+video_id, video_id=video_id)
+
+        if video_id.startswith('post/'):
+            content_subdict = downloaded_json['content']
+            url = content_subdict['contentUrl']
+            info_dict = {
+                'id': video_id,
+                'url': url,
+                'title': content_subdict.get('contentTitle'),
+                # This is a (potentially) shorter URL for the same video:
+                'webpage_url': self._RECOMMENDED_BASE_URL+video_id,
+                'manifest_url': url,
+                'is_live': False,
+                'was_live': False,
+                'live_status': 'not_live'
+            }
+
+            try:
+                info_dict['duration'] = int(content_subdict['duration'])
+            except:
+                pass
+            try:
+                info_dict['thumbnail'] = content_subdict['thumbnailUrl1']
+            except KeyError:
+                pass
+            try:
+                info_dict['description'] = content_subdict['contentDescription']
+            except KeyError:
+                pass
+            try:
+                info_dict['like_count'] = downloaded_json['likeCount']
+            except KeyError:
+                pass
+            try:
+                info_dict['dislike_count'] = downloaded_json['dislikeCount']
+            except KeyError:
+                pass
+            try:
+                info_dict['comment_count'] = downloaded_json['numComments']
+            except KeyError:
+                pass
+            try:
+                upload_date_time = downloaded_json['creationDateTime']
+            except KeyError:
+                pass
+            else:
+                info_dict['timestamp'] = unified_timestamp(upload_date_time)
+                info_dict['upload_date'] = unified_strdate(upload_date_time)
+            try:
+                created_by = downloaded_json['createdBy']
+            except KeyError:
+                pass
+            else:
+                try:
+                    info_dict['creator'] = created_by['name']
+                except KeyError:
+                    pass
+                try:
+                    info_dict['channel_id'] = created_by['id']
+                except KeyError:
+                    pass
+                try:
+                    info_dict['uploader_id'] = created_by['id']
+                except KeyError:
+                    pass
+                try:
+                    info_dict['channel'] = created_by['name']
+                except KeyError:
+                    pass
+
+            try:
+                video_availability = downloaded_json['premiumPlan']
+            except KeyError:
+                pass
```

Comment:
Read CONTRIBUTING.md on the correct way to do this

Reply:
I did, in fact, read CONTRIBUTING.md and _am_ aware of the **.get** method, if that's what you call "the correct way". (I am actually using .get in the code where there is no issue). Problem is: .get doesn't seem to work for these fields because initializing them with None (or other default value) would mean we are **fabricating** things. As an example, there should not be a "channel" field in our dictionary when no information regarding the channel was provided by the website.

Reply:
That sounds like a non-issue to me. It's not "fabricating" things to set something to a null value. But even if excluding the key completely was desired, it could be done like this:
```py
upload_date_time = downloaded_json.get('creationDateTime')

info_dict.update(k: v for (k, v) in {
    'duration': int_or_none(content_subdict['duration']),
    'thumbnail': content_subdict.get('thumbnailUrl1'),
    'description': content_subdict.get('contentDescription'),
    'like_count': downloaded_json.get('likeCount'),
    'dislike_count': downloaded_json.get('dislikeCount'),
    'comment_count': downloaded_json.get('numComments'),
    'timestamp': unified_timestamp(upload_date_time),
    'upload_date': unified_strdate(upload_date_time),
    'creator': try_get(downloaded_json, lambda x: x['createdBy']['name']),
    'channel_id': try_get(downloaded_json, lambda x: x['createdBy']['id'],
    'uploader_id': try_get(downloaded_json, lambda x: x['createdBy']['id'],
    'channel': try_get(downloaded_json, lambda x: x['createdBy']['name'],
}.items() if v is not None})
```

Reply:
there is no need to exclude the key completely. The entire YoutubeDL code is written by treating `None` as a lack of data, as pointed out in the docstring of `InfoExtractor`

Reply:
Fixed.